virtual/registry/types.go

```diff
@@ -65,6 +65,22 @@ type HeartbeatState struct {
 	Address string `json:"address"`
 }
 
+type refPhysicalState struct {
```

Comment:
I can understand your confusion regarding the necessity of introducing a separate struct instead of directly using the HeartbeatState. Allow me to explain my reasoning behind this:

1. While the server state currently aligns with the `Heartbeat`, it is crucial to consider future developments. We may introduce additional mechanisms to enhance the server state in the future. For example, we might implement a mechanism for tracking service availability or introducing a priority system for server selection. By having a dedicated struct, we ensure flexibility to accommodate these potential future enhancements without directly modifying the `HeartbeatState`.

2. The struct in question serves a specific purpose within the `dnsRegistry`. From what I gather from the code, the dnsRegistry does not store heartbeats anywhere.